src/pyqasm/cli/validate.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (C) 2024 qBraid
+#
+# This file is part of pyqasm
+#
+# Pyqasm is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for pyqasm, as per Section 15 of the GPL v3.
+
+"""
+Script to verify OpenQASM files
+
+"""
+
+import os
+from typing import Optional
+
+import typer
+from rich.console import Console
+
+from pyqasm import load
+
+
+def validate_paths_exist(paths: Optional[list[str]]) -> Optional[list[str]]:
+    """Verifies that each path in the provided list exists."""
+    if not paths:
+        return paths
+
+    non_existent_paths = [path for path in paths if not os.path.exists(path)]
+    if non_existent_paths:
+        if len(non_existent_paths) == 1:
+            raise typer.BadParameter(f"Path '{non_existent_paths[0]}' does not exist")
+
+        formatted_paths = ", ".join(f"'{item}'" for item in non_existent_paths)
+        raise typer.BadParameter(f"The following paths do not exist: {formatted_paths}")
+    return paths
+
+
+# pylint: disable-next=too-many-locals,too-many-statements
+def validate_qasm(src_paths: list[str], skip_files: Optional[list[str]] = None) -> None:
+    """Script validate OpenQASM files"""
+    skip_files = skip_files or []
+
+    failed_files: list[tuple[str, Exception]] = []
+
+    console = Console()
+
+    def should_skip(filepath: str, content: str) -> bool:
+        if filepath in skip_files:
+            return True
+
+        skip_tag = "// pyqasm: ignore"
+        line_number = 0
+
+        for line in content.splitlines():
+            line_number += 1
+            if 5 <= line_number <= 30 and skip_tag in line:
```

Comment:
How are `5` and `30` chosen here?

Reply:
No good reason, carried over from qbraid-cli headers check. Basically defines the bounds of where we check for for the skip comment. But can be removed, or require that file skip comment is on the first line.